### PR TITLE
Feature: Tickers calculation from influxdb

### PR DESCRIPTION
**Pull request by mnaichuk**
- Rewrite market_ticker daemon with fetching trades from Influxdb;
- Add `taker_type` to Trade model + set `taker_type` in migration;
- Rewrite Global `h24_volume` and `avg_h24_price` from influx;

### CODE REVIEW

File: db/migrate/20200228093038_add_taker_type_to_trade.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddTakerTypeToTrade < ActiveRecord::Migration[5.2]
+  def change
+    add_column :trades, :taker_type, :string, limit: 20, after: :taker_id
```
Comment (ysv) [inline db/migrate/20200228093038_add_taker_type_to_trade.rb]: update values for existing trades I think column should be not null

File: app/workers/amqp/market_ticker.rb [hunk 2]
```diff
@@ -11,49 +11,64 @@ def initialize
...
       def update_ticker(trade)
-        ticker        = @tickers[trade.market.id]
-        ticker[:low]  = get_market_low trade.market.id, trade
-        ticker[:high] = get_market_high trade.market.id, trade
-        ticker[:last] = trade.price
+        ticker_init   = Trade.from_influx_initialize_market_ticker(market: trade[:market_id])
+        ticker        = @tickers[trade[:market_id]]
+        ticker[:low]  = get_market_low trade[:market_id], trade
+        ticker[:high] = get_market_high trade[:market_id], trade
+        ticker[:open] = get_market_open trade[:market_id], ticker_init, trade
```
Comment (ysv) [inline app/workers/amqp/market_ticker.rb]: lazy load ticker_init if Rails cache expired

File: app/models/trade.rb [hunk 3]
```diff
@@ -53,10 +58,33 @@ def to_csv
     end
 
     def from_influx(params)
-      Peatio::InfluxDB.client.query("select id, price, amount, total, taker_type, market, time as created_at from trades where market='#{params['market']}' order by #{params['order_by']} limit #{params['limit']}") do |_name, _tags, points|
+      params.stringify_keys!
+      Peatio::InfluxDB.client.query("select id, price, amount, total, taker_type, market, created_at from trades where market='#{params['market']}' order by #{params['order_by']} limit #{params['limit']}") do |_name, _tags, points|
```
Comment (ysv) [inline app/models/trade.rb]: check queries cases

File: app/models/trade.rb [hunk 4]
```diff
@@ -53,10 +58,35 @@ def to_csv
...
+      end
+    end
+
+    # Average 24 hours price calculated using VWAP ratio.
+    def vwap_from_influx(params)
+      params.stringify_keys!
+      Peatio::InfluxDB.client.query("SELECT SUM(volume) / SUM(amount) AS vwap FROM (SELECT amount, price * amount AS volume FROM trades WHERE market='#{params['market']}' AND time > now() - 24h)") do |_name, _tags, points|
+        return points.map(&:deep_symbolize_keys!).first
+      end
+    end
```
Comment (mod) [inline app/models/trade.rb:89]: merge into one using total (volume)

**Pull request closed without merging**